libbeat/docs/processors-using.asciidoc

```diff
@@ -1035,3 +1036,77 @@ for each DNS request so if you have 2 nameservers then the total timeout will be
 `tag_on_failure`:: A list of tags to add to the event when any lookup fails. The
 tags are only added once even if multiple lookups fail. By default no tags are
 added upon failure.
+
+[[add-process-metadata]]
+=== Add process metadata
+
+The Add process metadata processor enriches events with information from running
+processes, identified by their process ID (PID).
+
+[source,yaml]
+-------------------------------------------------------------------------------
+processors:
+- add_process_metadata:
+    match_pids: [system.process.ppid]
+    target: system.process.parent
+-------------------------------------------------------------------------------
+
+The fields added to the event look as follow:
+-------------------------------------------------------------------------------
+"process": {
+  "name":  "systemd",
+  "title": "/usr/lib/systemd/systemd --switched-root --system --deserialize 22",
+  "exe":   "/usr/lib/systemd/systemd",
+  "args":  ["/usr/lib/systemd/systemd", "--switched-root", "--system", "--deserialize", "22"],
+  "pid:    1,
+  "ppid:   0,
+  "start_time": "2018-08-22T08:44:50.684Z",
+}
+-------------------------------------------------------------------------------
+
+Optionally, the process environment can be included, too:
+-------------------------------------------------------------------------------
+  ...
+  "env": {
+	"HOME":       "/",
+    "TERM":       "linux",
+    "BOOT_IMAGE": "/boot/vmlinuz-4.11.8-300.fc26.x86_64",
+    "LANG":       "en_US.UTF-8",
+  }
+  ...
+-------------------------------------------------------------------------------
+It has the following settings:
+
+`match_pids`:: List of fields to lookup for a PID. The processor will
+search the list sequentially until the field is found in the current event, and
+the PID lookup will be applied to the value of this field.
+
+`target`:: (Optional) Destination prefix where the `process` object will be
+created. The default is the event's root.
+
+`fields`:: (Optional) List of fields to add. By default, the processor will add
+all the available fields except `process.env`.
+```
+target: parent
+fields: [ process.title ]
+will create parent.process.title.
+```
+
+`ignore_missing`:: (Optional) When set to `false`, events that don't contain any
+of the fields in match_pids will be discarded and an error will be generated. By
+default, this condition is ignored.
+
+`ignore_errors`:: (Optional) When set to `true`, the event is not discarded when
+errors are encountered during enrichment.
+Possible errors include when a PID field cannot be converted to an integer or a
+field cannot be overwritten. It is still an error if no PID field is found. Use
+`ignore_missing` for that.
+
+`overwrite_fields`:: (Optional)
+By default, if a target field already exists, it will not be overwritten and an
+error will be logged. If `overwrite_fields` is set to `true`, this condition
+will be ignored.
+
+`restricted_fields`:: (Optional) By default, the `process.env` field is not
```

Comment:
This does not seem to be in line what the `restricted_fields` in the code?

Reply:
Maybe it's not well explained. If `false` process.env is not included (and errors if requested explicitly), it `true` is included. Is not what the code does?